brigade-worker/src/k8s.ts

```diff
@@ -442,6 +443,23 @@ export class JobRunner implements jobs.JobRunner {
       }
     }
 
+    // If the job defines volumes, add them to the pod's volume list.
+    // If the volume type is `hostPath`, first check if the project allows host mounts
+    // and throw an error if it it does not.
+    for (let v of job.volumes) {
+      if (v.hostPath != undefined && !project.allowHostMounts) {
+        throw new Error(`allowHostMounts is false in this project, not mounting ${v.hostPath.path}`);
+      }
+      this.runner.spec.volumes.push(v);
+    }
+
+    // If the job defines volume mounts, add them to every container's spec.
+    for (let m of job.volumeMounts) {
```

Comment:
Since `volumeMounts` and `volumes` appear to be un-coupled, should we add the same check here to be sure `project.allowHostMounts` is true?

And/or, in general, should we somehow check for proper API usage?  (If, say, configured with a volumeMount setup but no corresponding volume defined, etc. -- or is that allowed?)  I suppose as long as we can show the user the k8s library error that should result on mis-configuration, we'd be good.

Reply:
Since you need to declare both volume and volume mount, I think checking is unnecessary - it would end up just taking the volume mount name and iterating through the volumes to check if it's mounting a host path - which already do.

As for checking the existence of the volume you reference in the volume mount, that is a good idea, I'll add that.

Reply:
Added a check for the referenced volume, together with a test.

Reply:
Pushed `radumatei/brigade-worker:volume-mounts-check` if you need to try it on a project.

Reply:
I love the extra validation with `volumeExists` because that provides us a faster feedback loop 👍

Reply:
Perfect, thanks Radu!